### PR TITLE
[ADD] Add (Un)archive button on form view and add filter.

### PR DESCRIPTION
**Pull request by atp-odoo**
Task: https://www.odoo.com/web?#id=21106&view_type=form&model=project.task&action=327
Pad: https://pad.odoo.com/p/r.naBjPtiFyic2eja5

[ADD]: Add (Un)archive button on following model in form view and add filter also.
- slide.channel
- slide.slide
- forum.post
- forum.forum
- blog.blog
- blog.post
- survey.survey (only closed surveys)
- stock.location.route
- stock.warehouse.orderpoint
- stock.picking.type
- stock.incoterms
- stock.warehouse
- stock.location.path
- stock.location
- crm.team
- mrp.bom
- mrp.routing
- mrp.workcenter
- fleet.vehicle.log.contract
- fleet.vehicle
- event.track
- sales.claims
- hr.appraisals (only if done or cancelled)
- hr.equipment.request
- hr.equipment
- hr.department
- res.currency
- procurement.rule

### CODE REVIEW

File: addons/website_blog/models/website_blog.py [hunk 1]
```diff
@@ -19,8 +19,23 @@ class Blog(osv.Model):
...
+    }
+
+    def write(self, cr, uid, ids, vals, context=None):
+        res = super(Blog, self).write(cr, uid, ids, vals, context=context)
+        if 'active' in vals:
+            # archiving/unarchiving a blog does it on its posts, too
+            BlogPost = self.pool['blog.post']
+            ctx = dict(context, active_test=False)
+            posts = BlogPost.search(cr, uid, [('blog_id', 'in', ids)], context=ctx)
+            BlogPost.browse(cr, uid, posts, context=context).write({'active': vals['active']})
```
Comment (atp-odoo) [inline addons/website_blog/models/website_blog.py]: why need to browse then write? as is it old api write is not enough ? 
BlogPost.write(cr, uid, posts, {'active': vals['active']})

File: addons/fleet/fleet_view.xml [hunk 2]
```diff
@@ -530,6 +542,16 @@
             </field>
         </record>
 
+        <record id="fleet_vehicle_log_contract_view_search" model="ir.ui.view">
+            <field name="name">fleet.vehicle.log.contract.search</field>
+            <field name="model">fleet.vehicle.log.contract</field>
+            <field name="arch" type="xml">
+                <search string="Channel">
```
Comment (tivisse) [inline addons/fleet/fleet_view.xml]: Channel ?

File: addons/website_blog/models/website_blog.py [hunk 3]
```diff
@@ -20,8 +20,23 @@ class Blog(osv.Model):
...
         'name': fields.char('Blog Name', required=True, translate=True),
         'subtitle': fields.char('Blog Subtitle', translate=True),
+        'active': fields.boolean('Active'),
     }
 
+    _defaults = {
+        'active': True,
+    }
+
+    def write(self, cr, uid, ids, vals, context=None):
```
Comment (tivisse) [inline addons/website_blog/models/website_blog.py]: The write is not working in batch

Comment (tivisse) [inline addons/website_blog/models/website_blog.py]: Forget what I said

**Pull request merged**